libkbfs/interfaces.go

```diff
@@ -1225,6 +1233,31 @@ type DiskBlockCache interface {
 	Shutdown(ctx context.Context)
 }
 
+// DiskMDCache caches encrypted MD objects to the disk.
+type DiskMDCache interface {
+	// Get gets the latest cached MD for the given TLF from the disk cache.
+	Get(ctx context.Context, tlfID tlf.ID) (
+		buf []byte, ver kbfsmd.MetadataVer, localTimestamp time.Time, err error)
+	// Stage asks the disk cache to store the given MD in memory, but
+	// not yet write it to disk.  A later call to `Commit` or
+	// `Unstage` for `rev` or higher is required to avoid memory leaks.
+	Stage(ctx context.Context, tlfID tlf.ID, rev kbfsmd.Revision, buf []byte,
+		ver kbfsmd.MetadataVer, localTimestamp time.Time) error
+	// Commit writes a previously-staged MD to disk.  Trying to commit
+	// a revision that hasn't been staged is a no-op, to allow callers
+	// to call Commit without knowing whether Stage was called first
+	// (e.g., if the revision came from the cache in the first place).
+	Commit(ctx context.Context, tlfID tlf.ID, rev kbfsmd.Revision) error
```

Comment:
Like `Unstage` below, perhaps mention the behavior for committing when multiple revisions are staged here too? It seems only first one with given revision is committed.

Reply:
Also perhaps mention all revisions older than rev would be unstaged, like in your commit message.